doc/array.md

```diff
@@ -0,0 +1,78 @@
+# Array
+
+Class [`Napi::Array`][] inherits from class [`Napi::Object`][].
+
+Arrays are native representations of JavaScript Arrays. `Napi::Array` is a wrapper
+around `napi_value` representing a JavaScript Array.
+
+[`Napi::TypedArray`][] and [`Napi::ArrayBuffer`][] correspond to JavaScript data
+types such as [`Int32Array`][] and [`ArrayBuffer`][], respectively, that can be
```

Comment:
Missing the links for:
- `Napi::TypedArray`
- `Napi::ArrayBuffer`
- `Int32Array`
- `ArrayBuffer`